fundingmanager_test.go

```diff
@@ -1168,6 +1210,212 @@ func TestFundingManagerOfflinePeer(t *testing.T) {
 	assertNoChannelState(t, alice, bob, fundingOutPoint)
 }
 
+// TestFundingManagerPeerTimeoutAfterInitFunding checks that the zombie sweeper
+// will properly clean up a zombie reservation that times out after the
+// initFundingMsg has been handled.
+func TestFundingManagerPeerTimeoutAfterInitFunding(t *testing.T) {
+	alice, bob := setupFundingManagers(t)
+	defer tearDownFundingManagers(t, alice, bob)
+
+	// We will consume the channel updates as we go, so no buffering is needed.
+	updateChan := make(chan *lnrpc.OpenStatusUpdate)
+
+	// Create a funding request and start the workflow.
+	errChan := make(chan error, 1)
+	initReq := &openChanReq{
+		targetPubkey:    bob.privKey.PubKey(),
+		chainHash:       *activeNetParams.GenesisHash,
+		localFundingAmt: 500000,
+		pushAmt:         lnwire.NewMSatFromSatoshis(0),
+		private:         false,
+		updates:         updateChan,
+		err:             errChan,
+	}
+
+	alice.fundingMgr.initFundingWorkflow(bobAddr, initReq)
+
+	// Alice should have sent the OpenChannel message to Bob.
+	var aliceMsg lnwire.Message
```

Comment:
Why not use `assertFundingMsgSent` here?

Reply:
The reason I did not use `assertFundingMsgSent` here is because errors are handled differently in `initFundingWorkflow` so we must handle them differently here. Specifically, the `select` statement here as an additional `case` statement:

```
case err := <- initReq.err:
    t.Fatalf("error init funding workflow: %v", err)
```

Reply:
Makes sense, thanks!

Reply:
Can you check if your account is flagged by Travis? Seems like tests are not running. You might have to email them to get it unflagged.

Reply:
When that is done, you can rebase this on latest master :)

Reply:
Not sure why my account would be flagged but I'll check with them, thanks!